File: wallet-solution-mobile-app/remittance/src/main/java/com/ekenya/lamparam/utilities/PreferenceProvider.java
package com.ekenya.lamparam.utilities;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Preferences file class
 */
public class PreferenceProvider {

    private String ONBOARDING_DATA = "onboarding";
    private SharedPreferences preference;
    private String USER_NAME = "userName";

    public PreferenceProvider(Context context) {
        preference = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public void onboardUser(){
        preference.edit().putBoolean(ONBOARDING_DATA, true).apply();
    }

    public boolean userOnboarded(){
        return preference.getBoolean(ONBOARDING_DATA, false);
    }

    public void setUserName(String name){
        preference.edit().putString(USER_NAME, name).apply();
    }

    public String getUserName(){
        return preference.getString(USER_NAME, "Lennox Brown");
    }

}

File: app-lligrac/app/src/main/java/io/eclectics/cargilldigital/utils/PreferenceProvider.kt
package io.eclectics.cargilldigital.utils

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager


class PreferenceProvider(newBase: Context) {

    private val LANGUAGE_DATA = "language"

    private var appContext: Context? = null
    private var preference: SharedPreferences? = null

    init {
        appContext = newBase
        preference = PreferenceManager.getDefaultSharedPreferences(appContext)
    }

    fun setLanguage(language: String) {
        preference!!.edit().putString(LANGUAGE_DATA, language).apply()
    }

    fun getLanguage(): String {
        return preference!!.getString(LANGUAGE_DATA, "en")!!
    }


}